From b35a2c1552c910723bad5fcc21d3797b7c0ee9e3 Mon Sep 17 00:00:00 2001
From: Clark Wang <xiaoning.wang@nxp.com>
Date: Thu, 2 Jun 2022 15:33:25 +0800
Subject: [PATCH 151/274] MLK-25928 arm64: dts: imx93: add dma support for
 lpi2c and lpspi

Enable DMA mode for LPSPI and LPI2C instances.

NOTE:
For imx93 A0 chip, due to the limitation of edma2, lpspi3~8 do not
support using dma mode to transmit data over 32KB in one message.

Reviewed-by: Haibo Chen <haibo.chen@nxp.com>
Signed-off-by: Clark Wang <xiaoning.wang@nxp.com>
---
 arch/arm64/boot/dts/freescale/imx93.dtsi | 62 ++++++++++++++++++++++++
 1 file changed, 62 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/imx93.dtsi b/arch/arm64/boot/dts/freescale/imx93.dtsi
index 2917ae66c..4b8beeda2 100644
--- a/arch/arm64/boot/dts/freescale/imx93.dtsi
+++ b/arch/arm64/boot/dts/freescale/imx93.dtsi
@@ -245,6 +245,8 @@ lpi2c1: i2c@44340000 {
 				clocks = <&clk IMX93_CLK_LPI2C1_GATE>,
 					 <&clk IMX93_CLK_BUS_AON>;
 				clock-names = "per", "ipg";
+				dmas = <&edma1 7 0 0>, <&edma1 8 0 1>;
+				dma-names = "tx","rx";
 				status = "disabled";
 			};
 
@@ -255,6 +257,8 @@ lpi2c2: i2c@44350000 {
 				clocks = <&clk IMX93_CLK_LPI2C2_GATE>,
 					 <&clk IMX93_CLK_BUS_AON>;
 				clock-names = "per", "ipg";
+				dmas = <&edma1 9 0 0>, <&edma1 10 0 1>;
+				dma-names = "tx","rx";
 				status = "disabled";
 			};
 
@@ -267,6 +271,8 @@ lpspi1: spi@44360000 {
 				clocks = <&clk IMX93_CLK_LPSPI1_GATE>,
 					 <&clk IMX93_CLK_BUS_AON>;
 				clock-names = "per", "ipg";
+				dmas = <&edma1 11 0 0>, <&edma1 12 0 1>;
+				dma-names = "tx","rx";
 				status = "disabled";
 			};
 
@@ -279,6 +285,8 @@ lpspi2: spi@44370000 {
 				clocks = <&clk IMX93_CLK_LPSPI2_GATE>,
 					 <&clk IMX93_CLK_BUS_AON>;
 				clock-names = "per", "ipg";
+				dmas = <&edma1 13 0 0>, <&edma1 14 0 1>;
+				dma-names = "tx","rx";
 				status = "disabled";
 			};
 
@@ -621,6 +629,8 @@ lpi2c3: i2c@42530000 {
 				clocks = <&clk IMX93_CLK_LPI2C3_GATE>,
 					 <&clk IMX93_CLK_BUS_WAKEUP>;
 				clock-names = "per", "ipg";
+				dmas = <&edma2 8 0 0>, <&edma2 9 0 1>;
+				dma-names = "tx","rx";
 				status = "disabled";
 			};
 
@@ -631,9 +641,16 @@ lpi2c4: i2c@42540000 {
 				clocks = <&clk IMX93_CLK_LPI2C4_GATE>,
 					 <&clk IMX93_CLK_BUS_WAKEUP>;
 				clock-names = "per", "ipg";
+				dmas = <&edma2 10 0 0>, <&edma2 11 0 1>;
+				dma-names = "tx","rx";
 				status = "disabled";
 			};
 
+			/*
+			 * For imx93 A0 chip, due to the limitation of edma2,
+			 * lpspi3 does not support using dma mode to transmit
+			 * data over 32KB in one message.
+			 */
 			lpspi3: spi@42550000 {
 				#address-cells = <1>;
 				#size-cells = <0>;
@@ -643,9 +660,16 @@ lpspi3: spi@42550000 {
 				clocks = <&clk IMX93_CLK_LPSPI3_GATE>,
 					 <&clk IMX93_CLK_BUS_WAKEUP>;
 				clock-names = "per", "ipg";
+				dmas = <&edma2 12 0 0>, <&edma2 13 0 1>;
+				dma-names = "tx","rx";
 				status = "disabled";
 			};
 
+			/*
+			 * For imx93 A0 chip, due to the limitation of edma2,
+			 * lpspi4 does not support using dma mode to transmit
+			 * data over 32KB in one message.
+			 */
 			lpspi4: spi@42560000 {
 				#address-cells = <1>;
 				#size-cells = <0>;
@@ -655,6 +679,8 @@ lpspi4: spi@42560000 {
 				clocks = <&clk IMX93_CLK_LPSPI4_GATE>,
 					 <&clk IMX93_CLK_BUS_WAKEUP>;
 				clock-names = "per", "ipg";
+				dmas = <&edma2 14 0 0>, <&edma2 15 0 1>;
+				dma-names = "tx","rx";
 				status = "disabled";
 			};
 
@@ -787,6 +813,8 @@ lpi2c5: i2c@426b0000 {
 				clocks = <&clk IMX93_CLK_LPI2C5_GATE>,
 					 <&clk IMX93_CLK_BUS_WAKEUP>;
 				clock-names = "per", "ipg";
+				dmas = <&edma2 71 0 0>, <&edma2 72 0 1>;
+				dma-names = "tx","rx";
 				status = "disabled";
 			};
 
@@ -797,6 +825,8 @@ lpi2c6: i2c@426c0000 {
 				clocks = <&clk IMX93_CLK_LPI2C6_GATE>,
 					 <&clk IMX93_CLK_BUS_WAKEUP>;
 				clock-names = "per", "ipg";
+				dmas = <&edma2 73 0 0>, <&edma2 74 0 1>;
+				dma-names = "tx","rx";
 				status = "disabled";
 			};
 
@@ -807,6 +837,8 @@ lpi2c7: i2c@426d0000 {
 				clocks = <&clk IMX93_CLK_LPI2C7_GATE>,
 					 <&clk IMX93_CLK_BUS_WAKEUP>;
 				clock-names = "per", "ipg";
+				dmas = <&edma2 75 0 0>, <&edma2 76 0 1>;
+				dma-names = "tx","rx";
 				status = "disabled";
 			};
 
@@ -817,9 +849,16 @@ lpi2c8: i2c@426e0000 {
 				clocks = <&clk IMX93_CLK_LPI2C8_GATE>,
 					 <&clk IMX93_CLK_BUS_WAKEUP>;
 				clock-names = "per", "ipg";
+				dmas = <&edma2 77 0 0>, <&edma2 78 0 1>;
+				dma-names = "tx","rx";
 				status = "disabled";
 			};
 
+			/*
+			 * For imx93 A0 chip, due to the limitation of edma2,
+			 * lpspi5 does not support using dma mode to transmit
+			 * data over 32KB in one message.
+			 */
 			lpspi5: spi@426f0000 {
 				#address-cells = <1>;
 				#size-cells = <0>;
@@ -829,9 +868,16 @@ lpspi5: spi@426f0000 {
 				clocks = <&clk IMX93_CLK_LPSPI5_GATE>,
 					 <&clk IMX93_CLK_BUS_WAKEUP>;
 				clock-names = "per", "ipg";
+				dmas = <&edma2 79 0 0>, <&edma2 80 0 1>;
+				dma-names = "tx","rx";
 				status = "disabled";
 			};
 
+			/*
+			 * For imx93 A0 chip, due to the limitation of edma2,
+			 * lpspi6 does not support using dma mode to transmit
+			 * data over 32KB in one message.
+			 */
 			lpspi6: spi@42700000 {
 				#address-cells = <1>;
 				#size-cells = <0>;
@@ -841,9 +887,16 @@ lpspi6: spi@42700000 {
 				clocks = <&clk IMX93_CLK_LPSPI6_GATE>,
 					 <&clk IMX93_CLK_BUS_WAKEUP>;
 				clock-names = "per", "ipg";
+				dmas = <&edma2 81 0 0>, <&edma2 82 0 1>;
+				dma-names = "tx","rx";
 				status = "disabled";
 			};
 
+			/*
+			 * For imx93 A0 chip, due to the limitation of edma2,
+			 * lpspi7 does not support using dma mode to transmit
+			 * data over 32KB in one message.
+			 */
 			lpspi7: spi@42710000 {
 				#address-cells = <1>;
 				#size-cells = <0>;
@@ -853,9 +906,16 @@ lpspi7: spi@42710000 {
 				clocks = <&clk IMX93_CLK_LPSPI7_GATE>,
 					 <&clk IMX93_CLK_BUS_WAKEUP>;
 				clock-names = "per", "ipg";
+				dmas = <&edma2 83 0 0>, <&edma2 84 0 1>;
+				dma-names = "tx","rx";
 				status = "disabled";
 			};
 
+			/*
+			 * For imx93 A0 chip, due to the limitation of edma2,
+			 * lpspi8 does not support using dma mode to transmit
+			 * data over 32KB in one message.
+			 */
 			lpspi8: spi@42720000 {
 				#address-cells = <1>;
 				#size-cells = <0>;
@@ -865,6 +925,8 @@ lpspi8: spi@42720000 {
 				clocks = <&clk IMX93_CLK_LPSPI8_GATE>,
 					 <&clk IMX93_CLK_BUS_WAKEUP>;
 				clock-names = "per", "ipg";
+				dmas = <&edma2 85 0 0>, <&edma2 86 0 1>;
+				dma-names = "tx","rx";
 				status = "disabled";
 			};
 		};
-- 
2.25.1

